src/controllers/controllermanager.cpp

```diff
@@ -279,17 +303,45 @@ void ControllerManager::stopPolling() {
 }
 
 void ControllerManager::pollDevices() {
-    Trace tracer("ControllerManager::pollDevices");
-    bool eventsProcessed(false);
-    // Continue to poll while any device returned data.
-    do {
-        eventsProcessed = false;
-        foreach (Controller* pDevice, m_controllers) {
-            if (pDevice->isOpen() && pDevice->isPolling()) {
-                eventsProcessed = pDevice->poll() || eventsProcessed;
-            }
+    // Note: this function is called from a high priority thread which
+    // may stall the GUI or may reduce the available CPU time for other
+    // High Priority threads like caching reader or broadcasting more
+    // then desired, if it is called endless loop like.
+    //
+    // This especially happens if a controller like the 3x Speed
+    // Stanton SCS.1D emits more massages than Mixxx is able to handle
+    // or a controller like Hercules RMX2 goes wild. In such a case the
+    // receive buffer is stacked up every call to insane values > 500 messages.
+    //
+    // To avoid this we pick here a strategies similar like the audio
+    // thread. In case pollDevice() takes longer than a call cycle
+    // we are cooperative a skip the next cycle to free at least some
+    // CPU time
+    //
+    // Some random test data form a i5-3317U CPU @ 1.70GHz Running
+    // Ubuntu Trusty:
+    // * Idle poll: ~5 µs.
+    // * 5 messages burst (full midi bandwidth): ~872 µs.
+
+    if (m_skipPoll) {
```

Comment:
> It turns out that removing the while loop does not have any effect, if the buffer fills up because a pollDevices() call took more time than the poll Interval. In this case the timer expires during the pollDevices() and it is called immediately after it was finished.

Ahh ok -- thanks for the explanation.